src/polygonParts/models/polygonPartsManager.ts

```diff
@@ -174,28 +180,50 @@ export class PolygonPartsManager {
     );
   }
 
-  private buildFindPolygonPartsQuery({
-    clip,
-    polygonPartsEntityName,
-    footprint,
-    entityManager,
-  }: FindPolygonPartsOptions & { entityManager: EntityManager }): SelectQueryBuilder<FindPolygonPartsResponseItem> {
-    const canClip = !!footprint && clip;
-    const geometryField: PickPropertiesOfType<PolygonPartRecord, Geometry> = 'footprint';
-    const findPolygonPartsGeometryColumn = canClip
-      ? [`st_asgeojson((st_dump(st_intersection(${geometryField}, st_geomfromgeojson(:clipFootprint)))).geom, 15)::json as ${geometryField}`]
-      : [`st_asgeojson(${geometryField}, 15)::json as ${geometryField}`];
-    const findPolygonPartsColumns = [...this.findPolygonPartsColumns, ...findPolygonPartsGeometryColumn];
+  private buildFindPolygonPartsQuery(
+    context: FindPolygonPartsOptions & { entityManager: EntityManager }
+  ): SelectQueryBuilder<FindPolygonPartsQueryResponse> {
+    const { shouldClip, polygonPartsEntityName, footprint, entityManager } = context;
+    const canClip = !!footprint && shouldClip;
+    const geometryColumn = getMappedColumnName('footprint' satisfies PickPropertiesOfType<PolygonPartRecord, Geometry>);
 
     const polygonPart = entityManager.getRepository(PolygonPart);
     polygonPart.metadata.tablePath = polygonPartsEntityName.databaseObjectQualifiedName; // this approach may be unstable for other versions of typeorm - https://github.com/typeorm/typeorm/issues/4245#issuecomment-2134156283
 
-    const findQuery = polygonPart.createQueryBuilder('polygon_part').select(findPolygonPartsColumns);
-    return footprint
-      ? findQuery.where('st_intersects(footprint, st_geomfromgeojson(:clipFootprint))').setParameters({
-          clipFootprint: JSON.stringify(footprint),
-        })
-      : findQuery;
+    const clipCTE = polygonPart
+      .createQueryBuilder()
+      .select(
+        canClip
+          ? `st_asgeojson((st_dump(st_intersection(${geometryColumn}, st_geomfromgeojson(:clipFootprint)))).geom, 15)::jsonb`
```

Comment:
define the gridSize parameter as cofiguration default to 15

Reply:
added `maxDecimalDigits` like in the output of aggregation